cmd/payload-testing-prow-plugin/server.go

```diff
@@ -199,16 +212,23 @@ func (s *server) handle(l *logrus.Entry, ic github.IssueCommentEvent) string {
 				jobTuples = append(jobTuples, jobTuple)
 			}
 		}
+		specLogger.WithField("duration", time.Since(startResolveTests)).Debug("resolving tests completed")
 		if len(jobTuples) > 0 {
-			if err := s.kubeClient.Create(s.ctx, builder.build(jobTuples)); err != nil {
+			specLogger.Debug("creating PullRequestPayloadQualificationRuns ...")
+			startCreateRuns := time.Now()
+			run := builder.build(jobTuples)
+			runNames = append(runNames, run.Name)
+			if err := s.kubeClient.Create(s.ctx, run); err != nil {
 				specLogger.WithError(err).Error("could not create PullRequestPayloadQualificationRun")
 				return fmt.Sprintf("could not create PullRequestPayloadQualificationRun: %v", err)
 			}
+			specLogger.WithField("duration", time.Since(startCreateRuns)).WithField("runNames", runNames).Debug("creating PullRequestPayloadQualificationRuns completed")
 		} else {
 			specLogger.Warn("found no resolved tests")
 		}
 		messages = append(messages, message(spec, jobNames))
 	}
+	logger.WithField("duration", time.Since(start)).Debug("handle completed")
```

Comment:
Would it make sense to publish all these durations as metrics?

Reply:
I think yes if we have more profiling to come.
In that case, we just need to switch those duration to a histogram and expose it to prow-monitoring.